Content.Server/Cloning/CloningSystem.cs

```diff
@@ -38,32 +114,93 @@ internal void TransferMindToClone(Mind.Mind mind)
+        }
 
-            cloningPodComponent.UpdateStatus(CloningPodStatus.Cloning);
+        public bool IsPowered(CloningPodComponent clonepod)
+        {
+            if (!TryComp<ApcPowerReceiverComponent>(clonepod.Owner, out var receiver))
+                return false;
+
+            return receiver.Powered;
         }
 
-        public override void Update(float frameTime)
+        public bool TryCloning(EntityUid uid, Mind.Mind mind, HumanoidCharacterProfile hcp, CloningPodComponent? clonePod)
         {
-            foreach (var (cloning, power) in EntityManager.EntityQuery<CloningPodComponent, ApcPowerReceiverComponent>())
+            if (!Resolve(uid, ref clonePod))
+                return false;
+
+            if (clonePod.BodyContainer.ContainedEntity != null)
+                return false;
+
+            if (_cloningSystem.ClonesWaitingForMind.TryGetValue(mind, out var clone))
             {
-                if (cloning.UiKnownPowerState != power.Powered)
-                {
-                    // Must be *before* update
-                    cloning.UiKnownPowerState = power.Powered;
-                    UpdateUserInterface(cloning);
-                }
+                if (EntityManager.EntityExists(clone) &&
+                    TryComp<MobStateComponent>(clone, out var cloneState) &&
+                    !cloneState.IsDead() &&
+                    TryComp<MindComponent>(clone, out var cloneMindComp) &&
+                    (cloneMindComp.Mind == null || cloneMindComp.Mind == mind))
+                    return false; // Mind already has clone
+
+                _cloningSystem.ClonesWaitingForMind.Remove(mind);
+            }
+
+            if (mind.OwnedEntity != null &&
+                TryComp<MobStateComponent?>(mind.OwnedEntity.Value, out var state) &&
+                !state.IsDead())
+                return false; // Body controlled by mind is not dead
+
+            // Yes, we still need to track down the client because we need to open the Eui
+            if (mind.UserId == null || !_playerManager.TryGetSessionById(mind.UserId.Value, out var client))
+                return false; // If we can't track down the client, we can't offer transfer. That'd be quite bad.
+
+            if (!TryComp<TransformComponent>(clonePod.Owner, out var transform))
+                return false;
+
+            // Get species from player profile, this needs to get it from entity getting cloned instead
+            // This is currently the reason that someone can get scanned after being changed/chose ghost role and will be cloned
+            // as the one from their player profile.
+            // CHANGE THIS IN THE FUTURE TO GRAB DETAILS FROM SCANNED MOB
+            var speciesProto = _prototype.Index<SpeciesPrototype>(hcp.Species).Prototype;
+            var mob = Spawn(speciesProto, transform.MapPosition);
+            EntitySystem.Get<SharedHumanoidAppearanceSystem>().UpdateFromProfile(mob, hcp);
```

Comment:
Is this hard to do.

Also just cache the dependency.

Reply:
species is stored on the profile now so this is easy

Reply:
yep, thankfully species exists in appearance component now so able to take from there.